catalog/core/catalog-core-api/src/main/java/ddf/catalog/transform/QueryFilterTransformer.java

```diff
@@ -0,0 +1,38 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.transform;
+
+import ddf.catalog.operation.QueryRequest;
+
+/**
+ * An interface to transform {@link QueryRequest} containing external attribute names & values into a
+ * normalized taxonomy {@link QueryRequest}.
+ * <p>
+ * <b> This code is experimental. While this interface is functional and tested, it may change or be
+ * removed in a future version of the library. </b>
+ * </p>
+ */
+public interface QueryFilterTransformer {
+
+    /**
+     * Transforms a {@link QueryRequest} containing external attribute names & values into a
+     * normalized taxonomy {@link QueryRequest}. {@link QueryRequest} is used in order to allow
+     * the transformer to directly modify other properties (e.g. sourceId) based on criteria
+     * without requiring the endpoint to modify the request after conversion.  Transform method
+     * will use the FilterVisitor pattern and build the result Filter as the query is parsed.
+     *
+     * @return transformed {@link QueryRequest}
+     */
+    QueryRequest transform(QueryRequest queryRequest);
```

Comment:
In several other `Transformer` interfaces we include a properties map to allow additional information to be provided that may not be present in the object itself. I would strongly consider adding it to match the others.